tutorials/international/index.md

```diff
@@ -0,0 +1,281 @@
+you'll need to "internationalize" it. That means you'll need to write
+the app in a way that makes it possible to "localize" values like text
+and layouts for each language or "locale" that the app
+supports. Flutter provides widgets and classes that help with
+internationalization and the Flutter libraries themselves are
+internationalized.
+
+### The `Locale` class and the `Localizations` Widget
+
+The [`Locale`](https://docs.flutter.io/flutter/dart-ui/Locale-class.html)
+class is used to identify the user's language. Mobile devices support
+setting the locale for all applications, usually via a system settings
+menu. Internationalized apps respond by redisplaying values that are
+locale-specific. For example if the user switches the device's locale
+from English to French then a Text widget that displayed "Hello World"
+would be rebuilt with "Bonjour le monde".
+
+The
+[`Localizations`](https://docs.flutter.io/flutter/widgets/Localizations-class.html)
+widget defines the locale for its child and the localized resources
+that the child depends on. The
+[`WidgetsApp`](https://docs.flutter.io/flutter/widgets/WidgetsApp-class.html)
+widget creates a Localizations widget and rebuilds it if the system's locale changes.
+
+You can always lookup an app's current locale with `Localizations.localeOf()`:
+```
+Locale myLocale = Localizations.localeOf(context);
+```
+
+### Loading and Retrieving Localized Values
+
+The `Localizations` widget is used to load and lookup objects that
+contain collections of localized values. Apps refer to these objects
+with [`Localizations.of(context,type)`](https://docs.flutter.io/flutter/widgets/Localizations/of.html).
+If the device's locale changes, the Localizations widget automatically
+loads values for the new locale and then rebuilds widgets that used
+it. This happens because Localizations works like an
+[`InheritedWidget`](https://docs.flutter.io/flutter/widgets/InheritedWidget-class.html). When
+a build function refers to an inherited widget an implicit dependency
+on the inherited widget is created. When an inherited widget changes
+(when the Localizations widget's locale changes), its dependent
+contexts are rebuilt.
+
+Localized values are loaded by the Localizations widget's list of
+[`LocalizationsDelegate`](https://docs.flutter.io/flutter/widgets/LocalizationsDelegate-class.html)s. Each
+delegate must define an async
+[`load()`](https://docs.flutter.io/flutter/widgets/LocalizationsDelegate/load.html)
+method that produces an object which encapsulates a collection of
+localized values. Typically these objects define one method per localized value.
+
+In a large app, different modules or packages might be bundled with
+their own localizations. That's why the Localizations widget manages a
+table of objects, one per LocalizationsDelegate. To retrieve the
+object produced by one of the LocalizationsDelegate's `load` methods,
+you specify a BuildContext and the object's type.
```

Comment:
link BuildContext? (we seem to link most everything else :)

Reply:
I didn't because it's unlikely to be helpful.

https://docs.flutter.io/flutter/widgets/BuildContext-class.html

I guarantee no one reading about i18n will get though the first paragraph :-)